service/app/org/sunbird/telemetry/actor/TelemetryManagerActor.java

```diff
@@ -0,0 +1,181 @@
+import java.util.Map;
+
+import org.apache.commons.lang3.StringUtils;
+import org.sunbird.actor.core.BaseActor;
+import org.sunbird.actor.router.ActorConfig;
+import org.sunbird.common.exception.ProjectCommonException;
+import org.sunbird.common.models.response.Response;
+import org.sunbird.common.models.util.JsonKey;
+import org.sunbird.common.models.util.LoggerEnum;
+import org.sunbird.common.models.util.ProjectLogger;
+import org.sunbird.common.request.Request;
+import org.sunbird.common.responsecode.ResponseCode;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import telemetry.dispatcher.IDispatcher;
+import telemetry.dispatcher.TelemetryDispatcherFactory;
+
+/**
+ * 
+ * @author Mahesh Kumar Gangula
+ *
+ */
+
+@ActorConfig(tasks = { "dispatchTelemetry" }, asyncTasks = {})
+public class TelemetryManagerActor extends BaseActor {
+
+	List<String> dispatcherNames = new ArrayList<String>();
+	private static int defaultSize = 1000;
+	private static final String defaultDispacherName = "ekstep";
+	private ObjectMapper mapper = new ObjectMapper();
+
+	public TelemetryManagerActor() {
+		getMaxRequstSize();
+		getDispatchers();
+	}
+
+	private void dispatch(String dispatcherName, List<String> events) throws Exception {
+		IDispatcher dispatcher = TelemetryDispatcherFactory.getDispatcher(dispatcherName);
+		dispatcher.dispatch(events);
+	}
+
+	@Override
+	public void onReceive(Request request) throws Throwable {
+		String operation = request.getOperation();
+		if ("dispatchTelemetry".equals(operation)) {
+			List<String> events = getEvents(request.getRequest());
+			dispatch(defaultDispacherName, events);
+			for (String name : dispatcherNames) {
+				dispatch(name, events);
+			}
+			Response response = new Response();
+			response.put(JsonKey.RESPONSE, JsonKey.SUCCESS);
+			sender().tell(response, self());
+		} else {
+			onReceiveUnsupportedMessage(operation);
+		}
+	}
+
+	private List<String> getEvents(Map<String, Object> map) throws Exception {
+		String contentType = (String) map.get("content-type");
+		List<String> events = null;
+		switch (contentType) {
+		case "application/json":
+			Request request = (Request) map.get("body");
+			events = getEvents(request);
+			break;
+		case "application/zip":
+			byte[] body = (byte[]) map.get("body");
+			events = getEvents(body);
+			break;
+		default:
+			emptyRequestError("Please provide valid contnet-type.");
+		}
+		return events;
+	}
+
+	private void emptyRequestError(String message) throws ProjectCommonException {
+		throw new ProjectCommonException(ResponseCode.invalidRequestData.getErrorCode(), message,
+				ResponseCode.CLIENT_ERROR.getResponseCode());
+	}
+
+	private List<String> getEvents(Request request) throws Exception {
+		if (null == request)
+			emptyRequestError("Please provide valid request body. request body is empty.");
+
+		List<String> events = new ArrayList<String>();
+		if (request != null && request.getRequest() != null && !request.getRequest().isEmpty()) {
```

Comment:
Add helper method in Request for following check and use the same.

request != null && request.getRequest() != null && !request.getRequest().isEmpty()